src/components/structures/RoomView.js

```diff
@@ -340,7 +354,7 @@ module.exports = React.createClass({
         this.unmounted = true;
 
         // update the scroll map before we get unmounted
-        this._updateScrollMap(this.state.roomId);
+        RoomScrollStateStore.setScrollState(this.state.roomId, this._getScrollState());
```

Comment:
`this.state.roomId` may still be `null` here, in which case we shouldn't save the scroll state.